### PR TITLE
examples moved to seperate repo

### PR DESCRIPTION
**Pull request by Thirumurugan7**
The examples are removed from SDK repo and cloned from a different repo.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -77,21 +80,52 @@ function init(
...
-    // Copy Frontend Example
-    const sourceDirFrontend = path.join(
-        SDK_SOURCE_FOLDER_PATH,
-        "client-examples",
-        provingType == "ZK" ? "frontend_zk" : "frontend_opzk"
-    );
-    const destinationDirFrontend = path.join(destinationPath, "frontend");
-    copyFolderSync(sourceDirFrontend, destinationDirFrontend);
+    // Define the repository URL
+    const repoUrl = "https://github.com/zkspin/SDK-Examples.git";
```
[the remark this answers is not in the document]
Comment (Thirumurugan7) [reply src/index.ts]: completed

File: src/index.ts [hunk 2]
```diff
@@ -77,21 +80,52 @@ function init(
...
+        fs.renameSync(sourceDirGameplay, destinationDirGameplay);
+
+        // Remove the temporary directory
+        fs.rmSync(tempPath, { recursive: true, force: true });
+
+        logger.info(`Successfully initialized under folder: ${destinationPath}`);
+
+    } catch (error) {
+        throw error;
+    }
```
[the remark this answers is not in the document]
Comment (Thirumurugan7) [reply src/index.ts]: completed

File: src/index.ts [hunk 3]
```diff
@@ -77,21 +80,52 @@ function init(
...
+        if (provingType === "ZK") {
+            sourceDirFrontend = path.join(tempPath, 'client-examples', 'frontend_zk');
+        } else {
+            sourceDirFrontend = path.join(tempPath, 'client-examples', 'frontend_opzk');
+        }
+        const destinationDirFrontend = path.join(destinationPath, 'frontend');
 
-    logger.info(`Successfully initialized under folder: ${destinationPath}`);
+        // Ensure the frontend directory in the destination path is empty or does not exist
+        if (fs.existsSync(destinationDirFrontend)) {
```
[the remark this answers is not in the document]
Comment (Thirumurugan7) [reply src/index.ts]: completed

File: src/index.ts [hunk 4]
```diff
@@ -77,24 +46,64 @@ function init(
...
+
+  
+
+     
+ try {
+    await git.clone(repoConfig[provingType], tempPath);
+
+    // Determine which frontend directory to use based on provingType
+    const sourceDirFrontend = path.join(tempPath, `client-examples/frontend_${provingType.toLowerCase()}`);
+    const destinationDirFrontend = path.join(destinationPath, 'frontend');
```
Comment (tohacking) [inline src/index.ts:69]: I think we can rename the `frontend_opzk` or ` frontend_zk` repo here to just `frontent` and assume here just copy frontend directory.

File: src/index.ts [hunk 5]
```diff
@@ -77,24 +46,64 @@ function init(
...
+    fs.renameSync(sourceDirGameplay, destinationDirGameplay);
 
     logger.info(`Successfully initialized under folder: ${destinationPath}`);
+
+}  catch (error: unknown) {
+    if (error instanceof Error) {
+        logger.error(`Error during initialization: ${error.message}`);
+    } else {
+        logger.error(`An unknown error occurred: ${error}`);
+    }}
```
Comment (tohacking) [inline src/index.ts:96]: This line could be formatted

File: src/index.ts [hunk 6]
```diff
@@ -77,24 +46,64 @@ function init(
...
     // Copy Rust Gameplay Contract Example
-    const sourceDirGameplay = path.join(SDK_SOURCE_FOLDER_PATH, "gameplay");
-    const destinationDirGameplay = path.join(destinationPath, "gameplay");
-    copyFolderSync(sourceDirGameplay, destinationDirGameplay);
+    const sourceDirGameplay = path.join(tempPath, 'gameplay');
+    const destinationDirGameplay = path.join(destinationPath, 'gameplay');
+
+    // Ensure the gameplay directory does not exist
+    if (fs.existsSync(destinationDirGameplay)) {
+        logger.warn(`Gameplay directory ${destinationDirGameplay} already exists. Aborting.`);
```
Comment (tohacking) [inline src/index.ts:84]: I think an error message here would be better because `warning` typically means it's still functioning but a warning is issued.

**Pull request merged**